opcuaSource/README.md

```diff
@@ -0,0 +1,256 @@
+
+> Note: VANTIQ Extension Sources are first supported in VANTIQ version 1.23.
+
+## Extension Sources
+
+VANTIQ Extension Sources support the creation of a VANTIQ source.
+Within a VANTIQ system, a *source* is the means by which the VANTIQ system communicates with other systems.
+
+Briefly, extension sources connect to the VANTIQ system via a websocket connection, and interact with that system based on five (5) operations.
+
+ - connectExtension -- how the extension source identifies itself to VANTIQ
+ - configureExtension -- VANTIQ provides source-specific configuration to the extension source
+ - publish -- data sent from VANTIQ to the source
+   - Result of a PUBLISH to the source
+ - query -- queries sent from VANTIQ to the source with a response required.
+   - SELECT statements in VANTIQ
+ - notification -- data sent from the source to VANTIQ
+   - These appear as a message from the source
+
+More information about how to work with these operations can be found in the [Extension Source SDK for Java](../extjsdk/README.md) documentation.
+
+## OPC UA Extension Source Operation
+
+### Defining the Source in VANTIQ
+
+#### Creating the Source
+
+When creating an OPC UA Extension source,
+you must first create the OPC UA source type or implementation.
+This is done by using the `opcuaImpl.json` file found in `src/test/resources/opcuaImpl.json`.
+To make the source type known to VANTIQ, use the `vantiq` cli command
+
+```
+vantiq -s <profileName> load sourceimpls <fileName>`
+```
+
+where `<profileName>` is replaced by the VANTIQ profile name, and `<fileName>` is the file to be loaded.
+
+Once that type is loaded,
+you can create a source of that type.
+This is done by first selecting the OPCUA type for the source,
+
+![Selecting a type for a source](docs/images/opcuaExampleSelectType.png)
+
+and then providing a configuration.  We will examine the configuration just below.
+
+![Providing OPC UA Source Configuration](docs/images/opcuaExampleConfiguration.png)
+
+#### Configuration
+
+> Note:  It is assumed here that the reader is familiar with OPC UA concepts and terminology.
+> This source tries to use terminology familiar to an OPC UA practitioner where possible.
+
+The configuration of a VANTIQ extension source is a JSON document.
+
+For OPC UA sources, the document contains the following properties
+
+ - `discoveryEndpoint` -- This is the URL for the OPC UA server's discovery endpoint. This is the URL that the source will use to connect to the OPC UA server.
+ - `securityPolicy` -- This the specification (URI) of the security policy to be used to communicate with the OPC UA server. The security policy includes information about how (and whether) signing and encryption is done. The list of security policy URI's can be found in the OPC UA documentation.
+ - `monitoredItems` -- This is the list of items to be monitored by the OPC UA source.  More detail can be found below.
+
+#####<a id="monitored_items"></a>Monitored Items
```

Comment:
Displaying as `#####Monitored Items`. I'm guessing you just need to add a space.